evaluator/builtin_other.go

```diff
@@ -390,3 +390,19 @@ func unaryOpFactory(op opcode.Op) BuiltinFunc {
 		return
 	}
 }
+
+// CastFuncFactory produces builtin function according to field types.
+func CastFuncFactory(tp *types.FieldType) (BuiltinFunc, error) {
+	switch tp.Tp {
+	case mysql.TypeString, mysql.TypeDuration, mysql.TypeDatetime,
```

Comment:
Add comments for this check.
It is better to add the refer: http://dev.mysql.com/doc/refman/5.7/en/cast-functions.html